src/Catalyst.Cli/CommandTypes/BaseMessageCommand.cs

```diff
@@ -0,0 +1,147 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Concurrent;
+using System.Reactive.Linq;
+using Catalyst.Abstractions.Cli.Commands;
+using Catalyst.Abstractions.Cli.CommandTypes;
+using Catalyst.Abstractions.Cli.Options;
+using Catalyst.Abstractions.Rpc;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.IO.Events;
+using Catalyst.Core.Lib.IO.Messaging.Dto;
+using Catalyst.Protocol.Peer;
+using Google.Protobuf;
+using Serilog;
+
+namespace Catalyst.Cli.CommandTypes
+{
+    public abstract class BaseMessageCommand<TRequest, TResponse, TOption> : BaseCommand<TOption>,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fix this implementation of 'IDisposable' to conform to the dispose pattern.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)